### PR TITLE
PLATFORM-3211: unifying avatars and some images to use HTTPS protocol

### PR DESCRIPTION
**Pull request by harnash**
This is part of the effort of moving traffic to HTTPS. This changes avatars to use HTTPS as default. Additionally it changes `wfReplaceImageServer` to also use HTTPS which can affect other images served by MW.

See also: https://github.com/Wikia/config/pull/2292

ping: @Wikia/core-platform-team

### CODE REVIEW

File: extensions/wikia/UserProfilePageV3/Masthead.class.php [hunk 1]
```diff
@@ -435,7 +435,7 @@ public function isDefault() {
 		$url = $this->mUser->getGlobalAttribute( AVATAR_USER_OPTION_NAME );
 		if ( $url ) {
 			# all avatars (including the default ones) are always stored with a full URL
-			if ( startsWith( $url, 'http://' ) and preg_match('#/Avatar\d?.jpg$#', $url) ) {
+			if ( ( startsWith( $url, 'http://' ) or startsWith( $url, "https://" ) ) and preg_match('#/Avatar\d?.jpg$#', $url) ) {
```
Comment (WikiaTech) [inline extensions/wikia/UserProfilePageV3/Masthead.class.php]: MAJOR Replace "or" with "||". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2010)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by harnash** [comment]
One test case is failing due to incomplete config changes

File: extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ class UserProfilePageController extends WikiaController {
 	protected $title = null;
 
 	protected $defaultAvatars = null;
-	protected $defaultAvatarPath = 'http://images.wikia.com/messaging/images/';
+	protected $defaultAvatarPath = 'https://images.wikia.com/messaging/images/';
```
Comment (themech) [inline extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php]: Why do we change the path to `https://images.wikia.com/messaging/images/` and to `https://images.wikia.nocookie.net/messaging/images/` in `Masthead.class.php`?

Comment (harnash) [reply extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php]: @mech please see: ec816e50a01

**Review by themech — APPROVED** [review]

**Comment by harnash** [comment]
ping: @TK-999

File: extensions/wikia/UserProfilePageV3/Masthead.class.php [hunk 3]
```diff
@@ -100,6 +100,30 @@ public static function newFromUserName( $login ) {
...
+	}
+
+	/**
+	 * joinUrlPath -- utility function to properly concatenate url segments to avoid double '/'
+	 *
+	 * @param mixed -- any number of parameters which should be concatenated together
+	 *
+	 * @return string
+	 */
+	private static function joinUrlPath( ) {
```
Comment (mszabo-wikia) [inline extensions/wikia/UserProfilePageV3/Masthead.class.php]: Let's use varargs: `( string ...$segments )`

Comment (harnash) [reply extensions/wikia/UserProfilePageV3/Masthead.class.php]: Much better - thanks: 75eb0393c14

**Pull request merged**